### PR TITLE
Fix hit testing issues for wedge and annular wedge

### LINKED ISSUE #7428: Correctly showing Wedge Hover Tooltips with "clock" orient
#### ALL software version info (bokeh, python, notebook, OS, browser, any other relevant packages)

Python 3.6.3
Bokeh 0.12.10

#### Description of expected behavior and the observed behavior

The hover shows information of all the slices except the one you are pointing to, it should work the other way around.

#### Complete, minimal, self-contained example code that reproduces the issue

Example code taken from [https://stackoverflow.com/questions/48305017/correctly-showing-hover-tooltips-on-bokeh-wedge-chart](url)

```
import numpy as np
from bokeh.plotting import figure
from bokeh.io import show, output_file
from bokeh.models import HoverTool, ColumnDataSource
from math import pi



percents = [0, 5/143, 51/143, 88/143, 108/143, 141/143, 1.0]
category = ['A ', 'B ', 'C ', 'D ', 'E ', 'F']
counts = [5, 46, 37, 20, 33, 2]
starts = [1/2*pi-(p*2*pi) for p in percents[:-1]]
ends = [1/2*pi-(p*2*pi) for p in percents[1:]]
colors = ['#889dba', '#1f356f', '#1e92b8', '#33748a', '#a5d3e3', '#bbc2d4']
# create source
source = ColumnDataSource(
    data=dict(
        x=[0 for x in percents],
        y=[0 for x in percents],
        radius = [0.5 for x in percents],
        percents=percents,
        category= category,
        starts=starts,
        colors=colors,
        ends=ends,
        counts = counts
    )
)

TOOLS = "hover"

p = figure(plot_width = 500, plot_height = 500, x_axis_label = None, y_axis_label = None,
title = 'Type', tools = TOOLS)

p.title.align = 'center'
p.title.text_font = 'arial narrow'

p.wedge(x='x', y='y',  radius = 'radius', direction="clock",
                start_angle='starts', end_angle='ends', color='colors', source=source)

hover = p.select(dict(type=HoverTool))
hover.tooltips = [
    ('category', '@category'),
    ('percents','@counts')
]


p.axis.visible = False
p.ygrid.visible = False
p.xgrid.visible = False

output_file(pie.html')
show(p)

```

The hover works correctly if the user inverts the data and removes the clockwise statement as shown in this answer of the Stackoverflow question.
[https://stackoverflow.com/questions/48305017/correctly-showing-hover-tooltips-on-bokeh-wedge-chart/48306227#48306227](url)

### PR DESCRIPTION
**Pull request by anthonydouc**
Corrects the issues discussed in #7428 and #2080. Namely the incorrect behaviour when the direction is set to clock, and also improving hover tool tip display position for wedge glyphs.

Might be some changing required depending on the exact intention of using angle_between.

- [X] issues: fixes #7428 and #2080

### CODE REVIEW

File: bokehjs/src/lib/core/util/math.ts [hunk 1]
```diff
@@ -12,14 +12,15 @@ export function angle_dist(lhs: number, rhs: number): number {
   return Math.abs(angle_norm(lhs-rhs))
 }
 
-export function angle_between(mid: number, lhs: number, rhs: number, direction: "anticlock" | "clock"): boolean {
+export function angle_between(mid: number, lhs: number, rhs: number, direction: 1 | 0): boolean {
+  // 1 = anticlock, 0 = clock
```
Comment (mattpap) [inline bokehjs/src/lib/core/util/math.ts]: Use an enum to make the API more meaningful (https://www.typescriptlang.org/docs/handbook/enums.html).

Comment (anthonydouc) [reply bokehjs/src/lib/core/util/math.ts]: So its preferable to pass in a string with one of "anticlock" or "clock"? In that case I can change how the value is being passed into "angle_between"

Comment (bryevdv) [reply bokehjs/src/lib/core/util/math.ts]: @anthonydouc not strings, TypeScript supports actual enum variables, see e.g. https://github.com/bokeh/bokeh/blob/master/bokehjs/src/lib/models/tools/gestures/range_tool.ts#L11

**Comment by anthonydouc** [comment]
I must of missed that, that issue does not occur in every example depending on the width and height of the figure.

**Comment by bryevdv** [comment]
> I must of missed that, that issue does not occur in every example depending on the width and height of the figure. 

Yes, I noticed it even in the example above, different wedges would respond differently to hit testing. Easy to miss, and an example of why more eyes is good for OSS :D

**Comment by bryevdv** [comment]
brilliant, thanks @anthonydouc I am very happy to get this in the next release, it complements the `cumsum` transform addition, and together both make donut charts much more feasible

**Pull request merged**